File: backend/src/models/User.model.js
import mongoose from 'mongoose';

const userSchema = new mongoose.Schema({

    fullName: {
        type: String,
        required: [true, 'Full name is required'],
        trim: true,
    },

    email: {
        type: String,
        unique: true,
        trim: true,
        lowercase: true,
        match: [/^\S+@\S+\.\S+$/, 'Please enter valid email address'],
    },

    password: {
        type: String,
        required: [true, 'Password is required'],
        select: false,  // find() pe nhi dikhe ga
    },

    role: {
        type: String,
        enum: ['admin', 'receptionist', 'teacher', 'student'],
        required: [true, 'Role is required'],
        lowercase: true,
    },

    isActive: {  // still in Coa
        type: Boolean,
        default: true,
    }
}, { timestamps: true });

// userSchema.index({ email: 1 }); // Email ke basis par fast searching ke liye
// userSchema.index({ role: 1 }); // Role-wise filtering ke liye

export default mongoose.model('User', userSchema);

File: MobileApp/irp_mobile_app/lib/models/user_model.dart
class UserModel {
  final String id;
  final String fullName;
  final String email;
  final String role;
  final bool isActive;
  final String? profileType;
  final dynamic profile;

  UserModel({
    required this.id,
    required this.fullName,
    required this.email,
    required this.role,
    this.isActive = true,
    this.profileType,
    this.profile,
  });

  factory UserModel.fromJson(Map<String, dynamic> json) {
    return UserModel(
      id: json['id'] ?? json['_id'] ?? '',
      fullName: json['name'] ?? json['fullName'] ?? '',
      email: json['email'] ?? '',
      role: json['role'] ?? '',
      isActive: json['isActive'] ?? true,
      profileType: json['profileType'],
      profile: json['profile'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'fullName': fullName,
      'email': email,
      'role': role,
      'isActive': isActive,
      'profileType': profileType,
      'profile': profile,
    };
  }
}
